### PR TITLE
fix: CoreData tracking entity name retrieval

### LINKED ISSUE #2328: CoreData `TestEntity` error in test logs
### Description

In `SentryCoreDataTrackerTests` I see a system error being logged:

```CoreData: error: CoreData: error: Failed to call designated initializer on NSManagedObject class 'TestEntity'```

This could potentially cause problems down the line if we're calling/overriding alloc-init/new instead of alloc-initWithEntity.

### PR DESCRIPTION
**Pull request by armcknight**
Fixes #2328 and possible bug in CoreData tracker.

I originally noticed a system error logged for us using a non-designated initializer for an NSManagedObject subclass. Fixing that led to some test failures: 

image

which led to me discovering that we might have only ever tracked CoreData entity transactions using "NSManagedObject" in the span descriptions instead of the actual entity names being used by developers.

### CODE REVIEW

File: Sources/Sentry/SentryCoreDataTracker.m [hunk 1]
```diff
@@ -147,8 +147,8 @@ - (NSString *)descriptionForOperations:
 {
     NSMutableDictionary<NSString *, NSNumber *> *result = [NSMutableDictionary new];
 
-    for (id item in entities) {
-        NSString *cl = NSStringFromClass([item class]);
+    for (NSManagedObject *item in entities) {
+        NSString *cl = item.entity.name;
```
Comment (armcknight) [inline Sources/Sentry/SentryCoreDataTracker.m:151]: This is an actual functional change that I think was a bug.

**Review by brustolin — APPROVED** [review]
Thanks for this.
Looks like the bug is only during test. 
I run the old code in production and it gets the right name, but I believe your change will keep this future proof.

**Review by kevinrenskers — APPROVED** [review]

**Comment by armcknight** [comment]
Oh interesting, I wonder if that's because there's no xcdatamodel/NSManagedObjectModel in the unit tests @brustolin ? 🤔

**Comment by brustolin** [comment]
> Oh interesting, I wonder if that's because there's no xcdatamodel/NSManagedObjectModel in the unit tests @brustolin ? 🤔
> 
> If that's the case and there's no actual user-facing bug right now, i'll remove the changelog entry.

Im not sure if this is the reason. 
But is totally possible to have Core data in a project without the xcdatamodel.
No need to remove the changelog entry.

**Review by philipphofmann — APPROVED** [review]
Thanks.

**Pull request merged**